databuilder/databuilder/models/query/base.py

```diff
@@ -0,0 +1,69 @@
+from typing import Iterator
+
+from databuilder.models.graph_serializable import GraphSerializable
+
+
+class QueryBase(GraphSerializable):
+    @staticmethod
+    def _normalize(sql: str) -> str:
+        """
+        Normalizes a SQL query or SQL expression.
+
+        No checks are made to ensure that the input is valid SQL.
+        This is not a full normalization.  The following operations are preformed:
+
+        - Any run of whitespace characters outside of a quoted region is replaces by a single ' ' character.
+        - Characters outside of quoted regions are made lower case.
+        - If present, a trailing ';' is removed from the query.
+
+        Note:
+        Making characters outside quoted regions lower case does not in general result in an equivalent SQL statement.
+        For example, with MySQL the case sensitivity of table names is operating system dependant.
+        In practice, modern systems rarely rely on case sensitivity, and since making the non-quoted regions of the
+        query lowercase is very helpful in identifying queries, we go ahead and do so.
+
+        Also, this method fails to identify expressions such as `1 + 2` and `1+2`.
```

Comment:
I dont think this is something Amundsen needs to handle. The standardization below seems to be more than good enough for the overwhelming amount of use-cases